controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorCell.java

```diff
@@ -84,6 +85,8 @@ private class Delegate extends AbstractExecutionThreadService {
         private final EventProcessorConfig<T> eventProcessorConfig;
         private EventRead<T> event;
         private final CheckpointState state;
+        private Thread currentThread;
+        private AtomicBoolean interruptFlag = new AtomicBoolean(false);
```

Comment:
Replace these two with a single AtomicReference<Thread>. Then you can compare and set to null and interupt the returned thread. This way the member variable can be made final which we need for thread safety.

Reply:
Replaced the currentThread variable to `AtomicReference<Thread>` . Needed the interruptFlag to check the condition in triggerShutdown().

Reply:
Replaced the two with a single AtomicReference.